go/worker/compute/executor/committee/node.go

```diff
@@ -886,6 +891,8 @@ func (n *Node) Dispatch(batch transaction.RawBatch) error {
 	crash.Here(crashPointBatchPublishAfter)
 	spanPublish.Finish()
 
+	n.commonNode.CrossNode.Lock()
+	defer n.commonNode.CrossNode.Unlock()
```

Comment:
After re-taking the lock this should make sure that the round is still the same and that we are still in the correct state. Both may have changed while we were not holding the lock.